pkg/controller/stats/composite.go

```diff
@@ -0,0 +1,300 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stats
+
+import (
+	"bytes"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/exposure-notifications-verification-server/internal/icsv"
+	"github.com/google/exposure-notifications-verification-server/internal/project"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+	"github.com/google/exposure-notifications-verification-server/pkg/rbac"
+
+	keyserver "github.com/google/exposure-notifications-server/pkg/api/v1"
+)
+
+// Type assertions for CompositeStats
+var _ icsv.Marshaler = (CompositeStats)(nil)
+var _ json.Marshaler = (CompositeStats)(nil)
+
+// CompositeStats is an internal type for collecting unifed realm and key server stats.
+type CompositeStats []*CompositeDay
+
+// CompositeData represents a single day of composite stats.
+type CompositeDay struct {
+	Day            time.Time
+	RealmStats     *database.RealmStat
+	KeyServerStats *keyserver.StatsDay
+}
+
+type jsonCompositeStat struct {
+	RealmID uint                      `json:"realm_id"`
+	Stats   []*jsonCompositeStatStats `json:"statistics"`
+}
+
+type jsonCompositeStatStats struct {
+	Date time.Time                   `json:"date"`
+	Data *jsonCompositeStatStatsData `json:"data"`
+}
+
+type jsonCompositeStatStatsData struct {
+	// Fields that come from the realm stats
```

Comment:
Why not just embed both of those stats structs?

Reply:
I copied the structure here: https://github.com/google/exposure-notifications-verification-server/blob/main/pkg/database/realm_stats.go#L125

so that I could minimize javascript changes by having the exact same json structure

Reply:
If you move this into the database package though, couldn't you do the following?

```go
type jsonCompositeStatStatsData struct {
  jsonRealmStatStats
  // ...
}
```

I'm worried about things getting out of sync.

Reply:
dun. exported the one from the database, the other comes from the key server.